fastlane_core/lib/fastlane_core/configuration/configuration.rb

```diff
@@ -185,33 +205,22 @@ def load_configuration_file(config_file_name = nil, block_for_missing = nil, ski
     def fetch(key, ask: true)
       UI.user_error!("Key '#{key}' must be a symbol. Example :app_id.") unless key.kind_of?(Symbol)
 
-      option = option_for_key(key)
-      UI.user_error!("Could not find option for key :#{key}. Available keys: #{@available_options.collect(&:key).join(', ')}") unless option
+      option = verify_options_key!(key)
 
-      value = @values[key]
+      # Same order as https://docs.fastlane.tools/advanced/#priorities-of-parameters-and-options
+      value = if @values.key? key and !@values[key].nil?
```

Comment:
This is not only more readable, but more easily testable as well. By storing the config. file state on the Configuration object, you can mock a config. file by just setting the @config_file_options in tests and not even mocking the ConfigurationFile class.